### PR TITLE
fixed source files being dependent on *every* header

### PR DESCRIPTION
**Pull request by jytang**
was annoying, now it's not!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,14 +23,24 @@ else
...
 $(MAIN_TARGET): $(MAIN_OBJECTS)
 	$(CC) $(CFLAGS) $^ -o $@ $(LIBS)
 
-$(BUILD_DIR)/%.o: $(SRC_DIR)/%.cpp $(HEADERS)
+# Compilation of source files into object files, with help of depedency files.
+$(BUILD_DIR)/%.o: $(SRC_DIR)/%.cpp $(DEP_DIR)/%.dep
 	@mkdir -p $(dir $@)
 	$(CC) $(CFLAGS) $(INCS) -c $< -o $@
 
 clean:
```
Comment (metakirby5) [inline Makefile]: Can you move the `clean` rule to the bottom? :3

**Comment by metakirby5** [comment]
Looks like it works! Can you explain the dep thing?

File: Makefile (hunk 1 above)
Comment (jytang) [reply Makefile]: Done!

**Comment by jytang** [comment]
Yup. I use the compiler flag -MM to generate "depfiles" in the .dep directory.
If you look in to the files in the .dep directory you'll see that each one consists of its respective source file Make target with the headers it uses listed as dependencies.
For example in ``.dep/window.dep``:
```
.build/window.o: src/window.cpp src/window.h
```
``window.o`` only depends on one header file so it will only be recompiled if either of ``window.cpp`` or ``window.h`` is changed.

**Pull request merged**